Source/WebCore/Modules/webxr/WebXRWebGLLayer.cpp

```diff
@@ -228,19 +230,10 @@ ExceptionOr<RefPtr<WebXRViewport>> WebXRWebGLLayer::getViewport(WebXRView& view)
 
     auto& viewportData = view.eye() == XREye::Right ? m_rightViewportData : m_leftViewportData;
 
-    // 6. If the viewport modifiable flag is true and view’s requested viewport scale is not equal to current viewport scale:
-    //   6.1 Set current viewport scale to requested viewport scale.
-    //   6.2 Compute the scaled viewport.
-    if (view.isViewportModifiable() && view.requestedViewportScale() != viewportData.currentScale) {
-        viewportData.currentScale = view.requestedViewportScale();
-        m_viewportsDirty = true;
-    }
-
```

Comment:
I don't think it's okay to just delete all of this code.

Reply:
Why do we need the `m_viewportsDirty` boolean? It was not getting updated correctly, leading to viewports being stale and out of sync with `m_leftViewportData` and `m_rightViewportData`. Is it incorrect to always update `m_leftViewportData` and `m_rightViewportData`?